streakerbar/AppDelegate.swift

```diff
@@ -50,5 +51,40 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         updateTitle(String(arc4random_uniform(10)))
     }
     
+    func getGithubUsernameFromGitconfig() -> String {
+        let path = "~/.gitconfig"
+        let location = path.stringByExpandingTildeInPath
+        let data: NSData? = NSData(contentsOfFile: location)
+        
+        if let fileData = data {
+            let content = NSString(data: fileData, encoding:NSUTF8StringEncoding) as String
+            let matches = listMatches("user = \\w+", inString: content)
+            // if let is kinda weird but otherwise get Optional("chaserx")
+            if let username = replaceMatches("user = ", inString: matches[0], withString: "") {
+                println(username)
+                return username
+            }
+        }
+        return ""
```

Comment:
I might make this function return `String?` because you don't want to be asking GitHub for <empty string>'s events. If the result is nil, you can ask the user for their username manually.